src/expr/src/scalar/func.rs

```diff
@@ -818,6 +818,79 @@ fn convert_from<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError>
     }
 }
 
+fn encode<'a>(
+    data: Datum<'a>,
+    format: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let encoding = format.unwrap_str().to_lowercase();
+    let bytes = data.unwrap_bytes();
+    match encoding.as_str() {
+        "base64" => Ok(Datum::String(
+            temp_storage.push_string(base64::encode(bytes)),
+        )),
+        "hex" => Ok(Datum::String(temp_storage.push_string(hex::encode(bytes)))),
+        "escape" => {
+            let mut result = String::with_capacity(bytes.len());
+            for b in bytes {
+                if b == &('\\' as u8) {
+                    result += "\\\\";
+                } else if (b & 0x80) == 0x80 {
```

Comment:
Bytes with the high bit not set are also converted into escape sequences. In postgresql:

```
pstoev=# SELECT ENCODE('abc', 'escape');
 encode 
--------
 abc
(1 row)
```

In this branch:

```
materialize=> SELECT ENCODE('abc', 'escape');
 encode 
--------
 979899
(1 row)
```